src/main/java/com/couchbase/lite/BlobStore.java

```diff
@@ -555,30 +554,33 @@ public boolean storeBlob(byte[] data, BlobKey outKey) {
     }
 
     private static byte[] getBytesFromFile(File file) throws IOException {
-        InputStream is = new FileInputStream(file);
+        InputStream is = null;
+        try {
+            is = new FileInputStream(file);
 
-        // Get the size of the file
-        long length = file.length();
+            // Get the size of the file
+            long length = file.length();
 
-        // Create the byte array to hold the data
-        byte[] bytes = new byte[(int) length];
+            // Create the byte array to hold the data
+            byte[] bytes = new byte[(int) length];
```

Comment:
I'm not sure if this will return null or throw OOM exception. Probably this method needs rework. 

I believe you added this code in https://github.com/couchbase/couchbase-lite-java-core/commit/7e800559e9467b5173f89c8477b4ff68487f234e. Let handle this in a separated ticket.